src/index.ts

```diff
@@ -157,19 +162,19 @@ const create = <M extends Model>(
 
 const update = <M extends Model>(
   model: { new (): M } & typeof Model,
-  hooks?: WriteHooks
+  hooks?: UpdateHooks
 ): RequestHandler => async (req, res, next) => {
   try {
-    const record = await model.findByPk(req.params.id);
+    const record = await model.findByPk(req.params.id, { raw: true });
 
     if (!record) {
       return res.status(404).json({ error: "Record not found" });
     }
 
     const data =
-      hooks && hooks.before ? await hooks.before(req.body) : req.body;
+      hooks && hooks.before ? await hooks.before(req.body, record) : req.body;
 
-    await record.update(data);
+    await model.update(data, { where: { id: req.params.id } });
```

Comment:
Isn't possible to do record.update?

Reply:
Above `raw` is set to `true` so the model does not have the `update` method. I used it so that we only pass data in the hook. Otherwise users might generate some bugs by implementing side effects with these functions